### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by SamanAtashi**
# Create your own Enumerable
Features introduced are as follows:

- Use Ruby syntax for basic programming operations.
- Apply Ruby best practices and language style guides in code.

### CODE REVIEW

**Review by o-rashad — CHANGES_REQUESTED** [review]
Asslam Alikom team, @SamanAtashi @Leandro-Barretoo 

Hi 👋🏽, 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code

## Invalid Code Review Request

You have submitted a project, that doesn't fulfill one of the project **General requirements**  : 
- [x] Make sure that you used the correct workflow (Github flow). [LINK](https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/create_your_own_enumerable.md#:~:text=no%20linter%20errors.-,make,-sure%20that%20you)
  -  Ex: make your pull requests from the `feature` branch to the `main` branch.
  - Link on how easily you can change the pull request base branch [LINK](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Happy coding!👏:heart:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require_relative '../lib/enumerables'
+
+class MyList < MyEnumerable
```
Comment (helman101) [inline bin/main.rb]: - [x] MyEnumerable module should be included in MyList class, here MyList is inheriting from MyEnumerable, kindly change this.

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @SamanAtashi & @Leandro-Barretoo 👋, 

Good job so far with this project. 👏 👏👏
You've done well implementing some of the requirements, but there are still some which aren't addressed yet.

## Required Changes ♻️

- [x] _Check the comments under the review._

## Optional Changes 🤓 

These are all optional, but if you're interested in exploring this topic further, feel free to implement them. Any exploration here should be done outside program time.

- [ ] [OPTIONAL] If you decide to implement these requirements you should do it in a separate pull request. As always, remember to clearly document your decision in GitHub comments.

- For MyEnumerable implement:
    - #max
    - #min
    - #sort

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @SamanAtashi and @Leandro-Barretoo, 

All methods work well :clap: 
Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**